### PR TITLE
CF-503 Allow providing nova endpoint type

### PR DESCRIPTION
**Pull request by roman-verchikov**
In some cases `publicURL` used as nova endpoint type by default in nova
client returns empty list of services. This patch allows specifying nova
endpoint type for nova client to resolve these issues.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:808: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3632/

File: cloudferry/cfglib.py [hunk 1]
```diff
@@ -39,6 +39,11 @@
                help='Openstack region name'),
     cfg.StrOpt('service_tenant', default='service',
                help='Tenant name for services'),
+    cfg.StrOpt('nova_endpoint_type', default='publicURL',
```
Comment (antonf) [inline cloudferry/cfglib.py]: Please name it 'endpoint_type' and use it in other places where we create clients

**Comment by antonf** [comment]
Please fix failing unit tests

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:809: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.lib.os.compute.test_nova
tests/lib/os/compute/test_nova.py:422: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:425: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:427: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:456: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:460: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:462: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:491: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:495: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:497: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/lib/os/object_storage/swift_storage.py [hunk 2]
```diff
@@ -32,13 +32,15 @@ def get_swift_conn(self, params=None):
         if params is None:
             params = self.config.cloud
 
+        ep_type = {'endpoint_type': params.cloud.endpoint_type}
```
Comment (antonf) [inline cloudferry/lib/os/object_storage/swift_storage.py]: should be `params.endpoint_type`

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.os.compute.nova_compute
cloudferry/lib/os/compute/nova_compute.py:809: [E0602(undefined-variable), 
NovaCompute.change_status.<lambda>.<lambda>] Undefined variable 'status'
************* Module tests.lib.os.compute.test_nova
tests/lib/os/compute/test_nova.py:422: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:425: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:427: [W0212(protected-access), 
FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access 
to a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:455: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:456: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:460: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:462: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a 
protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:490: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _create_flavor_if_not_exists of a client class
tests/lib/os/compute/test_nova.py:491: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class
tests/lib/os/compute/test_nova.py:495: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _deploy_flavors of a client class
tests/lib/os/compute/test_nova.py:497: [W0212(protected-access), 
FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to 
a protected member _add_flavor_access_for_tenants of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
+1

**Comment by SVilgelm** [comment]
👍 can be merged

**Pull request merged**